Source/loadsave.cpp

```diff
@@ -865,9 +865,20 @@ void LoadItem(LoadHelper &file, Item &item)
 	GetItemFrm(item);
 }
 
-void LoadPremium(LoadHelper &file, int i)
+void LoadPremiumItems(LoadHelper &file)
 {
-	LoadAndValidateItemData(file, PremiumItems[i]);
+	// Resize the vector to the expected number of items in the save file
+	Blacksmith.items.resize(giNumberOfSmithPremiumItems);
+
+	for (int i = 0; i < giNumberOfSmithPremiumItems; ++i) {
+		LoadAndValidateItemData(file, Blacksmith.items[i]);
+	}
+
+	// Remove any empty/null items from the vector after loading
+	Blacksmith.items.erase(std::remove_if(Blacksmith.items.begin(), Blacksmith.items.end(), [](const Item &item) {
+		return item._itype == ItemType::None;
+	}),
+	    Blacksmith.items.end());

```

Comment:
Could there potentially be items that where not nulled and which where previously truncated by `PremiumItemCount`? Maybe we should just load the given number and then fast forward the point `(giNumberOfSmithPremiumItems - PremiumItemCount) * sizeof(Item)`